docs/DeveloperGuide.md

```diff
@@ -707,3 +707,214 @@ Coming soon
+seems to disappear.
+
+<img src="images/planned-enhancements/piechart-missing-label.png" alt="Screenshot of a pie chart with a large number of guests in the list" width="350px"> <br>
+
+This enhancement will improve the pie chart in WedLog by fixing these 2 issues.
+
+#### Benefits
+As a cosmetic enhancement, this will improve the overall look of the pie chart in WedLog. By ensuring labels appear when they need to and not appear when they should not, 
+this will allow users to better understand the pie chart, improving accessibility.
+
+#### Implementation
+The way we populate the pie chart data is by traversing the list of guests and allocating each pie chart slice with the RSVP status of each guest in order. Since we want to maintain a 
+specific colour scheme for the chart slices, we have to ensure that the order of slices is maintained. This can lead to unexpected behaviour such as the floating label appearing when there 
+are not yet guests with RSVP status "no". In JavaFX, in order to change the styling of the pie chart dynamically, we will need to change its style attributes in the controller class 
+**after** the scene has been shown. However, the current implementation of updating UI elements is done by removing existing components and re-rendering components when the data changes,
+i.e. after a command is executed. In particular, the pie chart is deleted and a new one is created based on the updated data. This means that it is not possible to conditionally style the 
+chart slices within the current implementation.
+
+One way to fix this issue is to change the way UI elements are updated. Instead of deleting and re-rendering components, we can update the existing components with the new data. 
+This is a non-trivial implementation and will require a rework of the UI code. This enhancement will then address the issues of (1) labels appearing when they should not, and (2) labels
+disappearing when they should not.
+
+<br>
+
+### 6. Better duplicate detection for tags
+#### Description
+Currently, tags are case-sensitive in WedLog. This means that if a user adds a tag with the same name but different case, e.g. 'family' and 'Family', WedLog will treat them as different tags. 
+This can lead to duplicate tags being added to WedLog, making it confusing for users. 
+
+<img src="images/planned-enhancements/duplicate-tags.png" alt="Screenshot of a guest card with duplicate tags"> <br>
+
+This enhancement will allow WedLog to detect duplicate tags, regardless of case. A tag will be stored in WedLog in lowercase, for standardisation.
+
+#### Benefits
+We understand that our users may assign tags with the same name but different case. This enhancement will enforce case-insensitivity for the input of tags and , 
+so as to prevent duplicate tags from being added to WedLog.
```

Comment:
```suggestion
We understand that our users may assign tags with the same name but different case. This enhancement will enforce case-insensitivity for the input of tags so as to prevent duplicate tags from being added to WedLog.
```